src/API/Management/Users.php

```diff
@@ -1,88 +1,265 @@
 <?php
-
+/**
+ * Users endpoints for the Management API.
+ *
+ * @package Auth0\SDK\API\Management
+ */
 namespace Auth0\SDK\API\Management;
 
-use Auth0\SDK\API\Header\ContentType;
-
-class Users extends GenericResource {
-
-    public function get($user_id) {
-
-        return $this->apiClient->get()
-            ->users($user_id)
+/**
+ * Class Users.
+ * Handles requests to the Users endpoint of the v2 Management API.
+ *
+ * @package Auth0\SDK\API\Management
+ */
+class Users extends GenericResource
+{
+    /**
+     * Get a single User by ID.
+     * Requires scopes: read:users, read:user_idp_tokens.
+     *
+     * @param string $user_id - User ID to get.
+     *
+     * @return mixed
+     *
+     * @throws \Exception
+     */
+    public function get($user_id)
+    {
+        return $this->apiClient->method('get')
+            ->addPath('users', $user_id)
             ->call();
     }
 
-    public function update($user_id, $data) {
-        return $this->apiClient->patch()
-            ->users($user_id)
-            ->withHeader(new ContentType('application/json'))
+    /**
+     * Update a User.
+     * Requires scope: update:users.
+     * May require scope: update:users_app_metadata
+     *
+     * @param string $user_id - User ID to update.
+     * @param array $data - User data to update:
+     *      - Only certain fields can be updated; see the @link below for allowed fields.
+     *      - "user_metadata" and "app_metadata" fields are merged, not replaced.
+     *
+     * @return mixed|string
+     *
+     * @throws \Exception
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/patch_users_by_id
+     */
+    public function update($user_id, $data)
+    {
+        return $this->apiClient->method('patch')
+            ->addPath('users', $user_id)
             ->withBody(json_encode($data))
             ->call();
     }
 
-    public function create($data) {
+    /**
+     * Create a new User.
+     * Requires scope: create:users.
+     *
+     * @param array $data - User create data:
+     *      - "connection" name field is required and limited to sms, email, & DB connections.
+     *      - "phone_number" field is required by sms connections.
+     *      - "email" field is required by email and DB connections.
+     *      - "password" field is required by DB connections.
+     *      - Depending on the connection user, other fields may be required.
```

Comment:
i think you meant `user->used` ??
Also, I don't think we enforce other fields apart from username. (We don't provide a way to customize this either AFAIK, maybe good to look at "custom db sign up scripts"). If it's that one alone, why not just name it?